src/interpreter/plugin/RomanPlugin.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * @license
+ * Copyright (c) 2020 Handsoncode. All rights reserved.
+ */
+
+import {CellError, ErrorType, InternalScalarValue, SimpleCellAddress} from '../../Cell'
+import {ErrorMessage} from '../../error-message'
+import {ProcedureAst} from '../../parser'
+import {ArgumentTypes, FunctionPlugin} from './FunctionPlugin'
+
+export class RomanPlugin extends FunctionPlugin {
+  public static implementedFunctions = {
+    'ROMAN': {
+      method: 'roman',
+      parameters: [
+        { argumentType: ArgumentTypes.NUMBER, minValue: 1, lessThan: 4000 },
+        { argumentType: ArgumentTypes.NOERROR, optionalArg: true, defaultValue: 0 }
+      ],
+    },
+    'ARABIC': {
+      method: 'arabic',
+      parameters: [
+        { argumentType: ArgumentTypes.STRING },
+      ],
+    },
+  }
+
+  public roman(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('ROMAN'),
+      (val: number, mode: InternalScalarValue) => {
+        val = Math.trunc(val)
+        if(mode === false) {
+          mode = 4
+        } else if(mode === true) {
+          mode = 0
+        }
+        mode = this.coerceScalarToNumberOrError(mode)
+        if(mode instanceof CellError) {
+          return mode
+        }
+        mode = Math.trunc(mode)
+        if(mode < 0) {
+          return new CellError(ErrorType.VALUE, ErrorMessage.ValueSmall)
+        }
+        if(mode > 4) {
+          return new CellError(ErrorType.VALUE, ErrorMessage.ValueLarge)
+        }
+        return romanMode(val, mode)
+      }
+    )
+  }
+
+  public arabic(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('ARABIC'),
+      (input: string) => {
+        input = input.trim().toUpperCase()
+        let minusSign = false
+        if(input.startsWith('-')) {
+          input = input.slice(1)
+          minusSign = true
+          if(input==='') {
+            return new CellError(ErrorType.VALUE, ErrorMessage.InvalidRoman)
+          }
+        }
+        let ans = 0;
+        [input, ans] = eatToken(input, ans,
+          {token: 'MMM', val: 3000},
+          {token: 'MM', val: 2000},
+          {token: 'M', val: 1000},
+          );
```

Comment:
Can you check the file for semicolons? It's not our current coding style (I don't agree with this rule but we should be consitent if we have it)

```suggestion
        )
```

Reply:
without semicolons, I getting TS compilation errors.

Reply:
```
src/interpreter/plugin/RomanPlugin.ts:66:24 - error TS2322: Type '[string, number]' is not assignable to type 'string | number'.
  Type '[string, number]' is not assignable to type 'string'.

```

Reply:
That's why I prefer semicolons :| 

Ok. Maybe we can refactor `eatToken` and `absorb` to consume an array instead of two arguments?

```ts
res = eatToken(res, ...)
```

As a bonus we would avoid destructuring it constantly.